test/powershell/Modules/Microsoft.PowerShell.Utility/Write-Stream.Tests.ps1

```diff
@@ -122,4 +125,119 @@ Describe "Stream writer tests" -Tags "CI" {
             $i.MessageData | Should -Be $null
         }
     }
+
+    Context 'Stream Variable Tests' {
+        # These tests validate that output from any stream can be captured in a variable
+
+        $streams = @('Error', 'Warning', 'Verbose', 'Debug', 'Information', 'Progress')
+        $streamTestCases = foreach ($stream in $streams) {
+            @{
+                Stream = $stream
+            }
+        }
+
+        BeforeAll {
+            function Test-StreamData {
+                [CmdletBinding()]
+                param()
+                Write-Progress -Activity 'Warming up' -PercentComplete ([Math]::Round(0 / 6 * 100))
+                Start-Sleep -Milliseconds 50
```

Comment:
Why do we need to sleep after every command? Can we sleep once at end of the function?
Also the sleep is not reliable on slow CIs - we should use our  helper function `Wait-Until`.

Reply:
I had sleep statements in that function so that I could actually see the progress window show up when running the tests interactively. That's not necessary for CI though, so I've removed them.